libs/designsystem/calendar/src/calendar.component.ts

```diff
@@ -434,10 +435,15 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     }
   }
 
-  _onDateSelected(newDay: CalendarCell) {
-    if (newDay.isSelectable && newDay.date) {
-      let newDate = new Date(this.activeMonth);
-      newDate.setDate(newDay.date);
+  _onDateSelected({ year, monthIndex, date, isSelectable }: CalendarCell) {
+    if (!isSelectable) return;
+
+    const isValidDate = date >= 1 && date <= 31;
+    const isValidMonth = monthIndex >= 0 && monthIndex <= 11;
+    const isValidYear = Number.isInteger(year);
+
+    if (isValidDate && isValidMonth && isValidYear) {
```

Comment:
Not sure why date (day/month/year) needs to be revalidated? I would assume the method for generating the days to render in the UI already yields valid dates? 🤔 
```suggestion
```